### PR TITLE
okta-aws-cli: init at version 1.0.1

### PR DESCRIPTION
**Pull request by daniyalsuri6**
###### Description of changes

- A CLI for having Okta as the IdP for AWS CLI operations
- Homepage ReadMe: https://github.com/okta/okta-aws-cli#readme

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by kalbasit — CHANGES_REQUESTED** [review]
Your commits should say:
- okta-aws-cli: init at version 1.0.1
- maintainers: add daniyalsuri

File: pkgs/tools/admin/okta-aws-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner  = "okta";
+    repo   = "okta-aws-cli";
+    rev    = "v${version}";
+    sha256 = "1qlc04fydcqnifbhrzdw63wq3ndzsyzp4s1bdvfhvizbhdi03mpp";
+  };
+
+  vendorSha256 = "1gzaadpf9zrci7yv88f434mvc7gralb7dj6wl6r4vsv3qk291680";
+
+  buildFlagsArray = [ "-ldflags=-s -w -X main.version=${version}" ];
```
[the remark this answers is not in the document]
Comment (daniyalsuri6) [reply pkgs/tools/admin/okta-aws-cli/default.nix]: done

File: pkgs/tools/admin/okta-aws-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
```
Comment (kalbasit) [inline pkgs/tools/admin/okta-aws-cli/default.nix:3]: The package seems to build many subpacakges but only installs one

```
Building subPackage ./cmd/okta-aws-cli
Building subPackage ./cmd/root
Building subPackage ./internal/agent
Building subPackage ./internal/ansi
Building subPackage ./internal/aws
Building subPackage ./internal/backoff
Building subPackage ./internal/config
Building subPackage ./internal/output
Building subPackage ./internal/sessiontoken
```

```
ls result/bin
okta-aws-cli
```

consider specifying 

```nix
subPackages = [ "cmd/okta-aws-cli" ];
```

to limit what it's building.

File: pkgs/tools/admin/okta-aws-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner  = "okta";
+    repo   = "okta-aws-cli";
+    rev    = "v${version}";
+    sha256 = "1qlc04fydcqnifbhrzdw63wq3ndzsyzp4s1bdvfhvizbhdi03mpp";
+  };
+
+  vendorSha256 = "1gzaadpf9zrci7yv88f434mvc7gralb7dj6wl6r4vsv3qk291680";
+
+  doCheck = false;
```
Comment (kalbasit) [inline pkgs/tools/admin/okta-aws-cli/default.nix:16]:
```suggestion
```

it doesn't seem to fail on my end, why the tests are disabled?

File: pkgs/tools/admin/okta-aws-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  ldflags = [ "-s" "-w" "-X=main.Version=${version}" ];
```
Comment (kalbasit) [inline pkgs/tools/admin/okta-aws-cli/default.nix:18]: I don't think this is working properly

```
 ./result/bin/okta-aws-cli --version
okta-aws-cli version 1.0.0
```

It's supposed to be 1.0.1.

It looks like upstream they have this configured in an [internal package as a const](https://github.com/okta/okta-aws-cli/blob/08b67cda2363c824c6a10ab9665f6844714cddda/internal/config/config.go#L35) which means we can't change it from the build process. I recommend you file a PR upstream to change this to a variable and update the package in nixpkgs at a later time. I can guide you privately on how to do that if you're unfamiliar with Go.

```suggestion
  ldflags = [ "-s" "-w" ];
```

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/okta-aws-cli/default.nix (hunk 3 above)
Comment (daniyalsuri6) [reply pkgs/tools/admin/okta-aws-cli/default.nix]: removed it

**Review by kalbasit — APPROVED** [review]

**Pull request merged**